src/h5web/visualizations/heatmap/AxisGrid.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState } from 'react';
+import { Dom, useThree, useFrame } from 'react-three-fiber';
+import { scaleLinear } from 'd3-scale';
+import IndexAxis from './IndexAxis';
+import styles from './HeatmapVis.module.css';
+
+interface AxisDomains {
+  left: [number, number];
+  bottom: [number, number];
+}
+
+interface Props {
+  dims: [number, number];
+  axisOffsets: [number, number];
+}
+
+function AxisGrid(props: Props): JSX.Element {
+  const { dims, axisOffsets } = props;
+  const [rows, cols] = dims;
+  const [leftAxisWidth, bottomAxisHeight] = axisOffsets;
+
+  const { camera, size } = useThree();
+  const { width, height } = size;
+
+  const [domains, setDomains] = useState<AxisDomains>({
+    left: [0, rows],
+    bottom: [0, cols],
+  });
+
+  // Axis bounds in R3F camera coordinates
+  const leftBounds: [number, number] = [
+    -height / 2 + bottomAxisHeight,
+    height / 2,
+  ];
+  const bottomBounds: [number, number] = [
+    -width / 2 + leftAxisWidth,
+    width / 2,
+  ];
+
+  // Scales R3F camera coordinates to axis bounds
+  const leftScale = scaleLinear()
+    .domain(leftBounds)
+    .range([0, rows]);
+  const bottomScale = scaleLinear()
+    .domain(bottomBounds)
+    .range([0, cols]);
+
+  useFrame(() => {
+    const { position, zoom } = camera;
+
+    // Scale bounds according to zoom and shift by camera position
+    setDomains({
+      left: leftBounds.map(b => leftScale(b / zoom + position.y)) as [
```

Comment:
Maybe introduce a type `Domain` for `[number, number]` to avoid this sort of wrapping. We're using it in a lot of places, so I think it makes sense.

Reply:
Et `bound` ou `val` au lieu de `b` ;)